### PR TITLE
PEP 517: Modernize pycti build

### PR DESCRIPTION
**Pull request by djds**
### Proposed changes

- [style] Update pre-commit hooks
- [typing] Add typing stubs to test-requirements.txt
- [build] PEP 517: Add pypa/build to test-requirements.txt
- [build] Add Makefile
- [build] Add pyproject.toml
- [build] Add setup.cfg
- [build] Remove setup.py
- [build] Set package version in __init__.py
- [ci.github] Autoformat codeql-analysis workflow config
- [ci.circleci] Autoformat CircleCI config
- [ci.circleci] PEP-517: Build deterministic wheel
- [ci.circleci] Add isort check, Python 3.7 tests
- [ci.circleci] Check version in CircleCI config
- [ci.circleci] Replace 'echo -e' with printf
- [ci.circleci] Update slack notification orb 4.7.1 => 4.8.0
- [ci.circleci] Parallelize tests for Python versions
- [style] Order imports with isort; autoformat with black
- [style] update black 20.8b1 => 22.1.0
- [connector] Lint opencti_connector_helper
- [ci.github] Lint markdown
- [doc] Lint requirements.txt
- [examples] Lint example code
- [tests] Add trailing newline to test data
- [style] Lint requirements.txt
- [style] Remove trailing whitespace

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Comment by nor3th** [comment]
Hey @djds 

Thank you very much for your amazing PR!!! Everything looks good to be, I'll wait for @SamuelHassine if he has any input on your PR before merging it.

Regards

**Review by nor3th — APPROVED** [review]

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    pytest-cases~=3.6
+    pytest-cov~=2.12
+    pytest_randomly~=3.8
+    types-python-dateutil
+    types-pytz
+    wheel~=0.36
+doc =
+    autoapi
+    sphinx-autodoc-typehints
+    sphinx_rtd_theme
```
Comment (nor3th) [inline setup.cfg:61]: Could you please add version numbers (either with >= or ~=) to the few python packages where none are defined

File: test-requirements.txt [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
 pre-commit~=2.13
-wheel~=0.36
 pytest~=6.2
+pytest-cases~=3.6
 pytest-cov~=2.12
-pytest-cases~=3.6
\ No newline at end of file
+pytest_randomly~=3.8
+types-python-dateutil
+types-pytz
```
Comment (nor3th) [inline test-requirements.txt:10]: Could you please add version numbers (either with >= or ~=) to the few python packages where none are defined

**Comment by SamuelHassine** [comment]
Good to me as well, you can merge @nor3th! Thanks a lot @djds for this amazing work!

**Review by SamuelHassine — APPROVED** [review]

**Pull request merged**